lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/persistence/DBOTeam.java

```diff
@@ -74,8 +77,24 @@ public Class<? extends DBOTeam> getDBOClass() {
 		}
 
 	};
-	
-	private static final MigratableTableTranslation<DBOTeam, DBOTeam> MIGRATION_MAPPER = new BasicMigratableTableTranslation<>();
+
+	@TemporaryCode(author = "peter.harvey@sagebase.org", comment = "One time migration of property canRequestMembership.  Can be removed after all teams have a canRequestMembership value.")
+	private static final MigratableTableTranslation<DBOTeam, DBOTeam> MIGRATION_MAPPER = new  MigratableTableTranslation<DBOTeam, DBOTeam>() {
+		@Override
+		public DBOTeam createDatabaseObjectFromBackup(DBOTeam backup) {
+			Team dto = TeamUtils.copyDboToDto(backup);

```

Comment:
Consider adding an enumeration (that is not exposed in the API), that translates the state of the two booleans into the state of the team E.g. (OPEN, PUBLIC, CLOSED) that is saved to the DB. Then instead of checking the booleans in the business logic we can work at the status level. Note that this status property also need to be migrated (but first we need to make sure we migrate the new requestMembership boolean).